extensions/ql-vscode/src/helpers.ts

```diff
@@ -29,8 +29,14 @@ export async function showAndLogErrorMessage(message: string, {
   items = [] as string[],
   fullMessage = undefined as (string | undefined)
 } = {}): Promise<string | undefined> {
-  return internalShowAndLog(message, items, outputLogger, Window.showErrorMessage, fullMessage);
+  const NUM_LINES_TO_KEEP = 2;
+  return internalShowAndLog(dropLinesExceptInitial(message, NUM_LINES_TO_KEEP), items, outputLogger, Window.showErrorMessage, fullMessage);
 }
+
+function dropLinesExceptInitial(message: string, n: number) {
```

Comment:
nit: We only ever use `2`, so this might be easier and you can avoid the constant. Not a big deal and either way is fine.

```suggestion
function dropLinesExceptInitial(message: string, n = 2) {
```